File: android/src/main/java/com/getcapacitor/community/admob/nativeads/NativeAdPluginEvents.java
package com.getcapacitor.community.admob.nativeads;

public enum NativeAdPluginEvents {
    LOADED("nativeAdLoaded"),
    FAILED_TO_LOAD("nativeAdFailedToLoad"),
    CLICKED("nativeAdClicked"),
    IMPRESSION("nativeAdImpression"),
    OPENED("nativeAdOpened"),
    CLOSED("nativeAdClosed"),
    AD_PAID("nativeAdPaid");

    private final String webEventName;

    NativeAdPluginEvents(String webEventName) {
        this.webEventName = webEventName;
    }

    public String getWebEventName() {
        return webEventName;
    }
}

File: ios/Sources/AdMobPlugin/NativeAds/NativeAdPluginEvents.swift
import Foundation

enum NativeAdPluginEvents: String {
    case loaded = "nativeAdLoaded"
    case failedToLoad = "nativeAdFailedToLoad"
    case clicked = "nativeAdClicked"
    case impression = "nativeAdImpression"
    case opened = "nativeAdOpened"
    case closed = "nativeAdClosed"
    case adPaid = "nativeAdPaid"
}
